src/bsoncxx/types/value.cpp

```diff
@@ -30,8 +30,6 @@ namespace types {
 #define BSONCXX_ENUM(name, val)                                                                \
     value::value(b_##name value) noexcept : _type(static_cast<bsoncxx::type>(val)),            \
                                             _b_##name(std::move(value)) {                      \
-        static_assert(std::is_nothrow_copy_constructible<b_##name>::value, "Copy may throw");  \
```

Comment:
This is very sad. I really don't want to drop these static asserts but boost::string_ref doesn't declare its copy ops noexcpet